### PR TITLE
[WIP] Vagrantfile added (fixes #45)

### LINKED ISSUE #45: merge ole--vagrant-treehouse and treehouse-builder?
merge this repo https://github.com/ole-vi/ole--vagrant-treehouses to current repo

### PR DESCRIPTION
**Pull request by dogi**
fixes #45

### CODE REVIEW

**Comment by paulbert** [comment]
Script isn't running on my Windows 10 machine now:
ole-vagrant

**Comment by mappuji** [comment]
Okay to be able to have this working we need to find the root cause why it failed in Windows 10.

Then fix it and test it on:
- [ ] Windows Machine
- [ ] Mac OSX
- [ ] Linux

**Comment by i5o** [comment]
For stretch I made another one https://github.com/ole-vi/ole--vagrant-treehouses/blob/vagrant2/Vagrantfile

**Comment by dogi** [comment]
Let me recap this PR works perfectly on macOS and Linux but fails on windows'es
since I am not a Windows user I can't fix it ...

**Comment by mappuji** [comment]
Should I reanimate this PR @dogi?

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Puppet, Chef, Ansible, Salt, and Docker are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  config.vm.provision "shell", inline: <<-SHELL
+    sudo apt update
+    sudo apt install -y kpartx qemu-user-static parted aria2
+    echo "git checkout <branch> ?"
+    mkdir -p /vagrant/images
+    sudo -u vagrant screen -dmS build sudo bash -c 'export PATH="$PATH:/sbin:/usr/sbin";cd /vagrant;./treehouse-builder --chroot'
```
Comment (mappuji) [inline Vagrantfile]: can we try to change `bash -c` to `sh -c` and try it again on the three platforms?

Comment (i5o) [reply Vagrantfile]: testing at this moment

**Comment by i5o** [comment]
Please test now. @mappuji  @dogi @paulbert

File: Vagrantfile (hunk 1 above)
Comment (mappuji) [reply Vagrantfile]: our windows might be only POSIX compliant but not support bash

**Pull request merged**